### ISSUE TITLE
Update announcement 

### ISSUE THREAD

**Issue by Ste74** [manjaro/release-plan#144]
Hi Team , @Chrysostomus do a good job with gnome-shell-extension-manjaro-update . 
This is a native gnome extension for notify update and after rework of @Chrysostomus this extension can manage also the msm , pamac and also notify at user if the upgrade need a manual intervent . Now when the extension is installed it disable the autostart of msm-notification and the pamac-tray service , only in Gnome but for work itredirect in pamac and in msm . At now the notify for user's manual intervent work but not efficiently .. if we want implement also in pamac it need rework in python and define how the announcement in a forum or in web page have to be formatted ( this in general ) .. @philmmanjaro , @guinux , @Kirek , @oberon2007 , @udeved .. 
Thinkings ?

**Comment by Chrysostomus** [comment]
The feature to notify about update announcements was requested by users in the forums. I believe that the feature would improve newbie friendliness of manjaro. There are pacman wrappers to do this for arch linux.

I implemented preliminary version of this in the gnome-shell-extension-manjaro-update. When checking for updates, it sends desktop notification that suggest reading announcements and running the update in terminal (with a button for this), if the last update in pacman log is older than the newest update announcements in the users branch.

This has the downside of creating too many false positives in stable branch, because it is triggered by any announcement for the users branch. It would be better to alert the user only about updates requiring manual intervention. There are 3 ways to accomplish this:

1) Tag the critical updates with some consistent message, such as [manual], [intervention], [terminal] or something. Or just * at the beginning of title would suffice too, just something I can grep.
2) Only announce the updates that require manual intervention. Currently announcement threads are used to discuss if updates had problems, so this would limit their functionality.
3) Someone suggest me a better way to parse the announcements than simply grepping the thread titles =)

If we want this kind of functionalityin other editions, we could integrate into the existing tools. Bspwm edition uses the same update-notifier as gnome, so it almost already has this, but the main editions are more important. There are some possibilities:

A) pacman hook that runs before update, checks for annoucements and asks for user input. On upside it could cover all editions, but it would be really annoing and difficult to handle with graphical frontends.
B) Integrate something similar to pamac. Someone could write something similar into update checker, or we could make it call a similar bash script that does the job. Former is cleaner, latter is easy.


But for any solution, we should probably tag the critical update annoucements, if we want to have something like this. Multiple users seemed to want it, and I think it makes sense.

**Comment by Chrysostomus** [comment]
The feature to notify about update announcements was requested by users in the forums. I believe that the feature would improve newbie friendliness of manjaro. There are pacman wrappers to do this for arch linux.

I implemented preliminary version of this in the gnome-shell-extension-manjaro-update. When checking for updates, it sends desktop notification that suggest reading announcements and running the update in terminal (with a button for this), if the last update in pacman log is older than the newest update announcements in the users branch.

This has the downside of creating too many false positives in stable branch, because it is triggered by any announcement for the users branch. It would be better to alert the user only about updates requiring manual intervention. There are 3 ways to accomplish this:

1) Tag the critical updates with some consistent message, such as [manual], [intervention], [terminal] or something. Or just * at the beginning of title would suffice too, just something I can grep.
2) Only announce the updates that require manual intervention. Currently announcement threads are used to discuss if updates had problems, so this would limit their functionality.
3) Someone suggest me a better way to parse the announcements than simply grepping the thread titles =)

If we want this kind of functionalityin other editions, we could integrate into the existing tools. Bspwm edition uses the same update-notifier as gnome, so it almost already has this, but the main editions are more important. There are some possibilities:

A) pacman hook that runs before update, checks for annoucements and asks for user input. On upside it could cover all editions, but it would be really annoing and difficult to handle with graphical frontends.
B) Integrate something similar to pamac. Someone could write something similar into update checker, or we could make it call a similar bash script that does the job. Former is cleaner, latter is easy.


But for any solution, we should probably tag the critical update annoucements, if we want to have something like this. Multiple users seemed to want it, and I think it makes sense.

**Comment by notramo** [comment]
What about making an RSS feed about the updates, that don't contains news about other things?
I think getting the news from forum titles is not good, because it's only in English language, and if Manjaro wants to be an user friendly distro, it must support translations.

This Pamac issue might be relevant: [https://github.com/manjaro/pamac/issues/215](url)

**Comment by Chrysostomus** [comment]
That would be better. 

To clarify, the script doesn't use any text from the titles, it only checks for a keyword to trigger announcement. Translation can be done separately. But still, dedicated rss feed is more useful.

**Comment by philmmanjaro** [comment]
Is this solved now?

**Comment by Ste74** [comment]
@Chrysostomus ?

**Comment by Chrysostomus** [comment]
Nope, I'm still waiting for info on if and how updates requiring manual intervention are announced. Tell me what to look for and I'll script it.

**Comment by Ste74** [comment]
https://github.com/manjaro/pamac/issues/281

**Comment by Ste74** [comment]
@Chrysostomus have you tried to source for rss feed this 
https://forum.manjaro.org/c/announcements.rss .. this is general but you can parse the branch with
https://forum.manjaro.org/c/announcements/stable-updates.rss 
for stable branch example
;)

**Comment by philmmanjaro** [comment]
@Ste74 , @Chrysostomus : any progress on this matter?

**Comment by Chrysostomus** [comment]
I'm still waiting for an answer. Can we have some kind of a tag for updates requiring manual update? If so, what it is? The code is trivial to do (mostly ready) once we agree on the trigger.

Alternatively it could trigger on any update announcement.

**Comment by papajoker** [comment]
for exemple for a invisible code, write : 
`    [i]&nbsp;[/i] optional text [i]&nbsp;[/i]`
generate (in `rss?)`
`    <span class="bbcode-i">&nbsp;</span>`

or simple or double char : `&#10;`

    &#10; optional text : requiring manual update &#10;
    &#10; &#10;
